subsys/bluetooth/audio/has_client.c

```diff
@@ -69,26 +90,210 @@ static struct has_inst *inst_by_conn(struct bt_conn *conn)
 	return NULL;
 }
 
-static void inst_cleanup(struct has_inst *inst)
+static struct bt_has_client *inst_find_by_addr(const bt_addr_le_t *addr)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(clients); i++) {
+		if (bt_addr_le_eq(&clients[i].addr, addr)) {
+			return &clients[i];
+		}
+	}
+
+	return NULL;
+}
```

Comment:
Given that you store the ID as well, don't we need to include the `id` when doing this lookup?

Reply:
Ditto for the `inst_alloc` below